### PR TITLE
feat: NFT details panel

### LINKED ISSUE #145: NFT details panel
A panel should be visible with the following data:

- Smart Contract ID: ct_2ibcu6C...d4
- Collection name: Test
- Owners: 10
- Amount: 10
- Extensions: "mintable", "mintable_limit", "burnable" (should be chips)
- Token limit: 10
- Template limit: 10

Token limit and template limit should be displayed based on the NFT extensions:
| Extension | Token limit | Template limit |
|--------|--------|--------|
| `mintable` without `mintable_limit` | Unlimited (∞ or n/a) | Not visible | 
| `mintable` & `mintable_limit` | `token_limit` value | Not visible | 
| `mintable_templates` without `mintable_templates_limit` | Not visible | Unlimited (∞ or n/a) | 
| `mintable_templates` & `mintable_templates_limit` | Not visible | `template_limit` value | 

Note:
- The standard currently only allows usage of `mintable` OR `mintable_templates`. It is not allowed to combine both.

Source: https://mainnet.aeternity.io/mdw/v2/aex141/ct_2w7MHbcyk5iM4yijZmjGcUF2DSZuVM1Mueppx5SSvPn5cgLJuZ

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #145

## Demo

https://github.com/aeternity/aescan/assets/15363559/6684cef1-d9de-4df8-8f03-b5daf5a64c7e

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] display nft details

### CODE REVIEW

File: src/pages/keyblocks/[id].vue [hunk 1]
```diff
@@ -13,7 +13,6 @@
 
   <template v-if="!isLoading">
     <keyblock-details-panel
-      v-if="keyblockDetails"
```
Comment (janmichek) [inline src/pages/keyblocks/[id].vue:16]: cleanup - handled by loader

File: src/utils/format.js [hunk 2]
```diff
@@ -84,6 +84,36 @@ export function formatDecodeByteArray(bytesArray) {
...
+  }
+}
+
+export function formatTemplateLimit(extensions, templateLimit) {
+  if (extensions.includes('mintable') && extensions.includes('mintable_limit')) {
+    return null
+  } else if (extensions.includes('mintable') || !extensions.includes('mintable_limit')) {
+    return null
+  } else if (extensions.includes('mintable_templates') && extensions.includes('mintable_templates_limit')) {
+    return 'Unlimited'
```
Comment (michele-franchi) [inline src/utils/format.js:105]:
```suggestion
    return templateLimit
```

Should return templateLimit, according to the table https://github.com/aeternity/aescan/issues/145

**Review by michele-franchi — CHANGES_REQUESTED** [review]
Good job, I added some suggestions.

File: src/utils/format.js [hunk 3]
```diff
@@ -84,6 +84,36 @@ export function formatDecodeByteArray(bytesArray) {
...
+  } else if (extensions.includes('mintable_templates') || !extensions.includes('mintable_templates_limit')) {
+    return templateLimit
```
[the remark this answers is not in the document]
Comment (janmichek) [reply src/utils/format.js:105]: fixed

File: src/utils/format.js (hunk 2 above)
Comment (janmichek) [reply src/utils/format.js]: fixed

**Review by lukeromanowicz — APPROVED** [review]
Very decent job :muscle: There is one detail missing but I approve in advance

File: src/pages/nfts/[id].vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+} catch (error) {
+  if ([400, 404].includes(error.response?.status)) {
+    throw showError({
+      data: {
+        entityId: route.params.id,
+        entityName: 'NFT',
+      },
+      statusMessage: 'EntityDetailsNotFound',
+    })
+  }
```
[the remark this answers is not in the document]
Comment (janmichek) [reply src/pages/nfts/[id].vue:45]: good catch, ser. Fixed

File: src/utils/hints/nftHints.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const nftsHints = {
+  nft: 'A non-fungible token (NFT) is a unit of data stored on a blockchain, that certifies a digital asset to be unique and therefore not interchangeable. NFTs can be used to represent items such as photos, videos, audio, and other types of digital files.',
+  collectionName: 'Name of the NFT collection.',
+  owners: 'The number of owners of NFTs collection.',
```
Comment (michele-franchi) [inline src/utils/hints/nftHints.js:4]:
```suggestion
  owners: 'Total number of owners of the NFT collection.',
```

File: src/utils/hints/nftHints.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+  amount: 'The number of NFTs in the collection.',
```
Comment (michele-franchi) [inline src/utils/hints/nftHints.js:5]:
```suggestion
  amount: 'Amount of NFTs in the collection.',
```

File: src/components/NftDetailsPanel.vue [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        :label="formatEllipseHash(nftDetails.contractId)"
+        :clipboard-text="nftDetails.contractId"
+        class="nft-details-panel__chip-ellipse"/>
+    </template>
+
+    <table>
+      <tbody>
+        <tr class="nft-details-panel__row">
+          <th class="nft-details-panel__table-header">
+            Collection name
```
Comment (michele-franchi) [inline src/components/NftDetailsPanel.vue:20]:
```suggestion
            Collection Name
```

**Review by michele-franchi — CHANGES_REQUESTED** [review]
I added some suggestions, some hints are not aligned to the NFTs list PR.

On mobile the Collection Name row goes to two lines, is it possible to fix it?
image

File: src/components/TokenDetailsPanel.vue [hunk 8]
```diff
@@ -236,11 +234,6 @@ const marketCap = computed(() =>
     }
   }
 
-  &__chip {
```
Comment (janmichek) [inline src/components/TokenDetailsPanel.vue:239]: clenup

File: src/utils/adapters.js [hunk 9]
```diff
@@ -1,14 +1,17 @@
...
+  formatAettosToAe,
+  formatBlockDiffAsDatetime,
+  formatDecodeBase64,
+  formatIsAuction,
+  formatNameStatus,
+  formatTemplateLimit,
+  formatTokenLimit,
+} from '@/utils/format'
 
-function isAuction(name) {
```
Comment (janmichek) [inline src/utils/adapters.js:7]: cleanup

**Comment by janmichek** [comment]
> I added some suggestions, some hints are not aligned to the NFTs list PR.

Now it is rebased

**Comment by janmichek** [comment]
> On mobile the Collection Name row goes to two lines, is it possible to fix it?

At the exact resolution I have differently broken content, but anyway, it's a more general issue. I am not really sure how to fix it in elegant way ./

**Review by michele-franchi — APPROVED** [review]
Looks good 👏

**Comment by CharisSiarampalis** [comment]
@janmichek Is the issue with the collection name going to get fixed on this pr?
Because i tried different mobile resolutions and it always appear on that spot
image

**Review by CharisSiarampalis — APPROVED** [review]

**Pull request merged**